packages/core/src/indexing/postgres/__tests__/postgres-index-api.test.ts

```diff
@@ -147,17 +161,52 @@ describe('indexStream', () => {
   test('new stream', async () => {
     const now = new Date()
     await indexApi.indexStream(STREAM_CONTENT)
-    const result: Array<any> = await dbConnection.from(`${MODELS_TO_INDEX[0]}`).select('*')
+    let result: Array<any> = await dbConnection
+      .from(`${MODELS_TO_INDEX[0]}`)
+      .select(
+        dbConnection.raw(
+          `*, stream_content->'settings'->'dark_mode' AS dark_mode, stream_content->>'id' AS id`
+        )
+      )
     expect(result.length).toEqual(1)
     const raw = result[0]
     expect(raw.stream_id).toEqual(STREAM_ID_B)
     expect(raw.controller_did).toEqual(CONTROLLER)
+    expect(raw.stream_content).toEqual(STREAM_TEST_DATA_PROFILE)
+    expect(raw.tip).toEqual(FAKE_CID.toString())
     expect(raw.last_anchored_at).toBeNull()
     expect(raw.first_anchored_at).toBeNull()
     const createdAt = new Date(raw.created_at)
     const updatedAt = new Date(raw.updated_at)
     expect(closeDates(createdAt, now)).toBeTruthy()
     expect(closeDates(updatedAt, now)).toBeTruthy()
+    expect(raw.dark_mode).toEqual(STREAM_TEST_DATA_PROFILE.settings.dark_mode)
+    expect(raw.id).toEqual(STREAM_TEST_DATA_PROFILE.id)
+
+    // create index on jsonb content
```

Comment:
can we break this into two tests?  I feel like we've conflated two test cases into one - first the basic test that the index gets maintained at all, then the test that the content is indexable and queryable with jsonb

Reply:
Yeah, I can push it up/split it into to and parse out the JSONB parts of it explicitly but I also feel we could leave it in one.

Since querying the content is about confirming that it was stored indeed as an object vs. a string, and therefor traversable, fits for me as part of the new stream creation/data insert test.

For the maintained index I see it's less about checking that the index is indeed maintained and more so as an extension of object storage and subsequent query-ability. I could see this only really fail if something would be severely wrong with the Postgres instance since we are just confirming base DB functionality via its built in operators at that point (which kinda makes the index test overkill in the first place thinking about it).

if we break it out I would rather do it as the db specific verifications then (namely jsonb / object related in the case of Postgres):
1. storage: is it stored correctly as an object confirmed via db operators
2. filtering: can I query against it with typescript data types being honored (string, bool etc.) without the need of addt. casting

and extend that then with e.g. MongoDB specific tests in the future.

Reply:
I don't feel that strongly about this.  IMO it'd be nice to split up, but we don't have to if you'd prefer not to.